### ISSUE TITLE
Why Not add a 3d library for it?

### ISSUE THREAD

**Issue by CleanCode-developer** [sam-astro/Z-Sharp#63]
Why Not add a 3d library for it?

**Comment by s5lachlan** [comment]
DNB Language.

![Image](https://github.com/user-attachments/assets/04747e75-fd34-4b10-ac43-2f51bbe0c463)

**Comment by dercode-solutions-2025** [comment]
I'm pretty sure no language has a 3D library, that's called a game engine

**Comment by s5lachlan** [comment]
I'm pretty sure no language has a 2D library, that's called a game engine

**Comment by dercode-solutions-2025** [comment]
> I'm pretty sure no language has a 2D library, that's called a game engine

Python has Pygame/TKinter/Turtle, C/C++ with SDL/SDL2/Vulkan/Olive.c, and lot's more.

**Comment by s5lachlan** [comment]
> > I'm pretty sure no language has a 2D library, that's called a game engine
> 
> Python has Pygame/TKinter/Turtle, C/C++ with SDL/SDL2/Vulkan/Olive.c, and lot's more.

Those are game engines.

**Comment by dercode-solutions-2025** [comment]
> > > I'm pretty sure no language has a 2D library, that's called a game engine
> > 
> > 
> > Python has Pygame/TKinter/Turtle, C/C++ with SDL/SDL2/Vulkan/Olive.c, and lot's more.
> 
> Those are game engines.

Those are libraries, not game engines, due to the fact they do not have a built-in IDE, No builtin GUI (You make GUIs with these libraries), and they are not full-fledged apps in the traditional sense as a game engine.

**Comment by s5lachlan** [comment]
> > > > I'm pretty sure no language has a 2D library, that's called a game engine
> > > 
> > > 
> > > Python has Pygame/TKinter/Turtle, C/C++ with SDL/SDL2/Vulkan/Olive.c, and lot's more.
> > 
> > 
> > Those are game engines.
> 
> Those are libraries, not game engines, due to the fact they do not have a built-in IDE, No builtin GUI (You make GUIs with these libraries), and they are not full-fledged apps in the traditional sense as a game engine.

So game engine = GUI?

**Comment by dercode-solutions-2025** [comment]
> > > > > I'm pretty sure no language has a 2D library, that's called a game engine
> > > > 
> > > > 
> > > > Python has Pygame/TKinter/Turtle, C/C++ with SDL/SDL2/Vulkan/Olive.c, and lot's more.
> > > 
> > > 
> > > Those are game engines.
> > 
> > 
> > Those are libraries, not game engines, due to the fact they do not have a built-in IDE, No builtin GUI (You make GUIs with these libraries), and they are not full-fledged apps in the traditional sense as a game engine.
> 
> So game engine = GUI?

No, it's that all game engines have GUIs and some do not need coding experience.

**Comment by s5lachlan** [comment]
> > > > > > I'm pretty sure no language has a 2D library, that's called a game engine
> > > > > 
> > > > > 
> > > > > Python has Pygame/TKinter/Turtle, C/C++ with SDL/SDL2/Vulkan/Olive.c, and lot's more.
> > > > 
> > > > 
> > > > Those are game engines.
> > > 
> > > 
> > > Those are libraries, not game engines, due to the fact they do not have a built-in IDE, No builtin GUI (You make GUIs with these libraries), and they are not full-fledged apps in the traditional sense as a game engine.
> > 
> > 
> > So game engine = GUI?
> 
> No, it's that all game engines have GUIs and some do not need coding experience.

Grok is my favourite game engine.
